### PR TITLE
[DevCenter] fixing generated samples

### PR DESCRIPTION
**Pull request by sebrenna**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/devcenter/Azure.Developer.DevCenter/samples/Sample_CreateDeleteDevBoxAsync.md [hunk 1]
```diff
@@ -1,5 +1,68 @@
+# Azure.Developer.DevCenter samples - Dev Boxes
+
 To use these samples, you'll first need to set up resources. See [getting started](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/devcenter/Azure.Developer.DevCenter/README.md#getting-started) for details.
 
-## <scenario> Asynchronously create, connect to, and delete a Dev Box
+## Get all projects in a dev center
```
Comment (pallavit) [inline sdk/devcenter/Azure.Developer.DevCenter/samples/Sample_CreateDeleteDevBoxAsync.md:5]: I am assuming these samples are directly coming from the test/samples directory.

Comment (sebrenna) [reply sdk/devcenter/Azure.Developer.DevCenter/samples/Sample_CreateDeleteDevBoxAsync.md:5]: Yes - we just hadn't wired those samples into these markdown files yet.

This is to resolve a comment on this Arch Board issue: https://github.com/Azure/azure-sdk/issues/5723

**Comment by pallavit** [comment]
LGTM.

**Review by pallavit — APPROVED** [review]

File: sdk/devcenter/Azure.Developer.DevCenter/samples/Sample_CreateDeleteEnvironmentAsync.md [hunk 2]
```diff
@@ -1,5 +1,73 @@
...
+
+Create a `DevCenterClient` and issue a request to get all projects the signed-in user can access.
+
+```C# Snippet:Azure_DevCenter_GetProjects_Scenario
+var credential = new DefaultAzureCredential();
+var devCenterClient = new DevCenterClient(endpoint, credential);
+string targetProjectName = null;
+await foreach (BinaryData data in devCenterClient.GetProjectsAsync(filter: null, maxCount: 1))
+{
+    JsonElement result = JsonDocument.Parse(data.ToStream()).RootElement;
```
[the remark this answers is not in the document]
Comment (pallavit) [reply sdk/devcenter/Azure.Developer.DevCenter/samples/Sample_CreateDeleteEnvironmentAsync.md:15]: Yes, totally. This makes me wonder if we have other usages of JsonDocument.Parse in SDK samples we should proactively try. Do we want to pursue this change before DynamicJson GA or after?

**Pull request merged**